contracts/engine/GenArt721MinterDAExp_PBAB.sol

```diff
@@ -0,0 +1,611 @@
+    // `onlyWhitelisted()` method on the associated core contract
+    modifier onlyCoreAllowlisted() {
+        require(
+            genArtCoreContract.isWhitelisted(msg.sender),
+            "Only Core allowlisted"
+        );
+        _;
+    }
+
+    // modifier to restrict access to only the artist for a given project
+    modifier onlyArtist(uint256 _projectId) {
+        require(
+            (msg.sender ==
+                genArtCoreContract.projectIdToArtistAddress(_projectId)),
+            "Only Artist"
+        );
+        _;
+    }
+
+    // modifier to restrict access to calls only involving a valid projectId
+    // (an existing project)
+    modifier onlyValidProjectId(uint256 _projectId) {
+        require(
+            _projectId < genArtCoreContract.nextProjectId(),
+            "Only existing projects"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract to be integrated with
+     * Art Blocks core contract at address `_genArt721Address`.
+     * @param _genArt721Address Art Blocks core contract address for
+     * which this contract will be a minter.
+     */
+    constructor(address _genArt721Address) ReentrancyGuard() {
+        genArt721CoreAddress = _genArt721Address;
+        genArtCoreContract = IGenArt721CoreV2_PBAB(_genArt721Address);
+    }
+
+    /**
+     * @notice Sets the minter owner (the platform provider) address to `_ownerAddress`.
+     * @param _ownerAddress New owner address.
+     */
+    function setOwnerAddress(
+        address payable _ownerAddress
+    ) public onlyCoreAllowlisted {
+        ownerAddress = _ownerAddress;
+    }
+
+    /**
+     * @notice Sets the minter owner (the platform provider) revenue % to `_ownerPercentage` percent.
+     * @param _ownerPercentage New owner percentage.
+     */
+    function setOwnerPercentage(
+        uint256 _ownerPercentage
+    ) public onlyCoreAllowlisted {
+        ownerPercentage = _ownerPercentage;
+    }
+
+    /**
+     * @notice Syncs local maximum invocations of project `_projectId` based on
+     * the value currently defined in the core contract. Only used for gas
+     * optimization of mints after maxInvocations has been reached.
+     * @param _projectId Project ID to set the maximum invocations for.
+     * @dev this enables gas reduction after maxInvocations have been reached -
+     * core contracts shall still enforce a maxInvocation check during mint.
+     * @dev function is intentionally not gated to any specific access control;
+     * it only syncs a local state variable to the core contract's state.
+     */
+    function setProjectMaxInvocations(uint256 _projectId) external {
+        uint256 maxInvocations;
+        (, , , maxInvocations, , , , , ) = genArtCoreContract.projectTokenInfo(
+            _projectId
+        );
+        // update storage with results
+        projectConfig[_projectId].maxInvocations = uint24(maxInvocations);
```

Comment:
I think we want the following for PBAB minter, since core max invocations may be increased (see `GenArt721Minter_PBAB.sol`:
```
        uint256 maxInvocations;
        uint256 invocations;
        (, , invocations, maxInvocations, , , , , ) = genArtCoreContract
            .projectTokenInfo(_projectId);
        // update storage with results
        projectMaxInvocations[_projectId] = maxInvocations;
        // update locally stored project max invocations has been reached
        projectConfig[_projectId].maxHasBeenInvoked = (invocations == maxInvocations)
```

we also gate this function to core whitelisted on the current PBAB minter, but I don't see a major reason to add that gate (since max invocations is ultimately controlled by the core, and this is just a "sync" to that value).

Reply:
Updating logic here to:

```
    function setProjectMaxInvocations(uint256 _projectId) external {
        uint256 maxInvocations;
        uint256 invocations;
        (, , invocations, maxInvocations, , , , , ) = genArtCoreContract.projectTokenInfo(
            _projectId
        );
        // update storage with results
        projectConfig[_projectId].maxInvocations = uint24(maxInvocations);
        if (invocations < maxInvocations) {
            projectConfig[_projectId].maxHasBeenInvoked = false;
        }
    }
    ```